File: AutoSlider/ViewController.m
//
//  ViewController.m
//  AutoSlider
//
//  Created by Kshitij Ghule on 09/04/17.
//  Copyright © 2017 Crystal Hitech IT Solutions Pvt. Ltd. All rights reserved.
//


#import "ViewController.h"
#import "TAPageControl.h"


@interface ViewController () <UIScrollViewDelegate, TAPageControlDelegate>
{
    NSTimer *timer;
    NSInteger index;
}
@property (strong, nonatomic) TAPageControl *customPageControl2;
@property (weak, nonatomic) IBOutlet UIScrollView *scrollView;
@property (strong, nonatomic) NSArray *imagesData;

@end

@implementation ViewController
#pragma mark - Lifecycle
- (void)viewDidLoad
{
    [super viewDidLoad];
    
    self.imagesData = @[@"image1.jpg", @"image2.jpg", @"image3.jpg"];
    for(int i=0; i<self.imagesData.count;i++){
        UIImageView *imageView = [[UIImageView alloc] initWithFrame:CGRectMake(CGRectGetWidth(self.view.frame) * i, 0, CGRectGetWidth(self.view.frame), CGRectGetHeight(self.scrollView.frame))];
        imageView.contentMode = UIViewContentModeScaleAspectFill;
        imageView.image = [UIImage imageNamed:[self.imagesData objectAtIndex:i]];
        [self.scrollView addSubview:imageView];
    }
    self.scrollView.delegate = self;
    index=0;
    
    
    // Progammatically init a TAPageControl with a custom dot view.
    self.customPageControl2 = [[TAPageControl alloc] initWithFrame:CGRectMake(20,self.scrollView.frame.origin.y+self.scrollView.frame.size.height,self.scrollView.frame.size.width,40)];//CGRectMake(0, CGRectGetMaxY(self.scrollView.frame) - 100, CGRectGetWidth(self.scrollView.frame), 40)
    // Example for touch bullet event
    self.customPageControl2.delegate      = self;
    self.customPageControl2.numberOfPages = self.imagesData.count;
    self.customPageControl2.dotSize       = CGSizeMake(20, 20);
    self.scrollView.contentSize = CGSizeMake(CGRectGetWidth(self.view.frame) * self.imagesData.count, CGRectGetHeight(self.scrollView.frame));
    [self.view addSubview:self.customPageControl2];
    
}

-(void)viewDidAppear:(BOOL)animated{
    timer=[NSTimer scheduledTimerWithTimeInterval:3 target:self selector:@selector(runImages) userInfo:nil repeats:YES];
}

-(void)viewDidDisappear:(BOOL)animated{
    if (timer) {
        [timer invalidate];
        timer=nil;
    }
}

-(void)runImages{
    self.customPageControl2.currentPage=index;
    if (index==self.imagesData.count-1) {
        index=0;
    }else{
        index++;
    }
    [self TAPageControl:self.customPageControl2 didSelectPageAtIndex:index];
}

- (void)viewDidLayoutSubviews
{
    [super viewDidLayoutSubviews];
    
    
}

#pragma mark - ScrollView delegate
- (void)scrollViewDidScroll:(UIScrollView *)scrollView
{
    NSInteger pageIndex = scrollView.contentOffset.x / CGRectGetWidth(scrollView.frame);
    self.customPageControl2.currentPage = pageIndex;
    index=pageIndex;
}
// Example of use of delegate for second scroll view to respond to bullet touch event
- (void)TAPageControl:(TAPageControl *)pageControl didSelectPageAtIndex:(NSInteger)currentIndex
{
    index=currentIndex;
    [self.scrollView scrollRectToVisible:CGRectMake(CGRectGetWidth(self.view.frame) * currentIndex, 0, CGRectGetWidth(self.view.frame), CGRectGetHeight(self.scrollView.frame)) animated:YES];
}


@end

File: AutoSlider/ViewController.swift
//
//  ViewController.swift
//  AutoSlider
//
//  Created by Avijit Nagare on 29/04/18.
//  Copyright © 2018 Crystal Hitech IT Solutions Pvt. Ltd. All rights reserved.
//

import UIKit

class ViewController: UIViewController,UIScrollViewDelegate,TAPageControlDelegate {
  
    var index = 0
    var timer = Timer()
    
    var customPageControl2 = TAPageControl()
    @IBOutlet weak var scrollView: UIScrollView!
    var imageData = NSArray()
    var lblArray = NSArray()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.imageData = ["image1.jpg", "image2.jpg","image3.jpg"]
        ////Add label on Image
        self.lblArray = ["Forest","Sunshine","Sea"]
        
        for i in 0..<self.imageData.count {
            
            let xPos =  self.view.frame.size.width * CGFloat(i)
            let imageView = UIImageView(frame: CGRect(x:xPos, y: 0, width: self.view.frame.width, height: self.scrollView.frame.size.height))
            imageView.contentMode = .scaleAspectFill
            imageView.image = UIImage(named: self.imageData[i] as! String)
            
            //Add label on Image
//            let lbl = UILabel()
//            lbl.text = String("\(self.lblArray[i] as! String)")
//            lbl.textColor = UIColor.green
//            lbl.sizeToFit()
//            lbl.center = CGPoint(x: imageView.frame.size.width  / 2,
//                                 y: imageView.frame.size.height / 2);
//            imageView.addSubview(lbl)
         
            self.scrollView.addSubview(imageView)
        }
         self.scrollView.delegate = self
        index=0
        
        self.customPageControl2 = TAPageControl(frame: CGRect(x: 20, y: self.scrollView.frame.origin.y+self.scrollView.frame.size.height, width: self.scrollView.frame.size.width, height: 40))
        self.customPageControl2.delegate = self
        self.customPageControl2.numberOfPages =  self.imageData.count
        self.customPageControl2.dotSize = CGSize(width: 20, height: 20)
        
        self.scrollView.contentSize = CGSize(width: self.view.frame.size.width * CGFloat(self.imageData.count), height: self.scrollView.frame.size.height)
        self.view.addSubview(self.customPageControl2)
        
        
        // Do any additional setup after loading the view.
    }
    
    override func viewDidLayoutSubviews() {
        super.viewDidLayoutSubviews()
    }
    
    override func viewDidAppear(_ animated: Bool) {
        timer = Timer.scheduledTimer(timeInterval: 3, target: self, selector: #selector(runImages), userInfo:nil, repeats: true)
    }
    override func viewDidDisappear(_ animated: Bool) {
        timer.invalidate()
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    func runImages() {
        self.customPageControl2.currentPage = index
        if index == self.imageData.count - 1 {
            index=0
        }else{
            index = index + 1
        }
        self.taPageControl(self.customPageControl2, didSelectPageAt: index)
    }
    
    func scrollViewDidScroll(_ scrollView: UIScrollView) {
        let pageIndex = scrollView.contentOffset.x / scrollView.frame.size.width
        self.customPageControl2.currentPage = Int(pageIndex)
        index = Int(pageIndex)
    }
    func taPageControl(_ pageControl: TAPageControl!, didSelectPageAt currentIndex: Int) {
        index = currentIndex
        self.scrollView.scrollRectToVisible(CGRect(x: self.view.frame.size.width * CGFloat(currentIndex), y: 0, width: self.view.frame.width, height: self.scrollView.frame.size.height), animated: true)
    }
    
    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        // Get the new view controller using segue.destinationViewController.
        // Pass the selected object to the new view controller.
    }
    */

}
